omnibus/files/private-chef-cookbooks/private-chef/recipes/nginx.rb

```diff
@@ -169,6 +169,25 @@
   end
 end
 
+if node['private_chef']['opscode-erchef']['stats_user']
+  stats_api_passwd = PrivateChef.credentials.get('opscode_erchef', 'stats_password')
+  stats_passwd_file = node['private_chef']['opscode-erchef']['stats_password_file']
+
+  execute 'stats_api_append_password' do
+    command "openssl passwd -apr1 '#{stats_api_passwd}' >> #{stats_passwd_file}"
+    action :nothing
+    sensitive true
+  end
+
+  file stats_passwd_file do
+    content "#{node['private_chef']['opscode-erchef']['stats_user']}:"
+    mode '0400'
+    owner OmnibusHelper.new(node).ownership['owner']
+    group OmnibusHelper.new(node).ownership['group']
+    notifies :run, 'execute[stats_api_append_password]', :immediately
+  end
+end
```

Comment:
As of the veil work, we try not to put any passwords outside of the veil store when the user has `insecure_addon_compat false`.  We've never really talked about how strict we want to be about that moving forward since that work was done to address some particular customer concerns.

Unfortunately, from the docs it looks like it might be hard/impossible to use the basic_auth module with out sticking the credential in that file.  I suppose we could change over to using access_by_lua, but that seems a bit complex for this.

Reply:
The password is salted and hashed from my understanding.

```
cat /var/opt/opscode/nginx/stats_htpasswd
foo:$apr1$S.a9NOEq$YiWDrTTJb68HjyFjm9CmH/```

So it might be ok?

Reply:
http://nginx.org/en/docs/http/ngx_http_auth_basic_module.html does seem to mention that the actual algorithms used might be...  weak?  I get suspicious when I see MD5 in there.  But there seems to be no other nice way to do this.

Reply:
@ksubrama Ah, yeah, somehow I overlooked the fact that this is in fact NOT a "plain text password" in that file.  I think that means we are OK.